pkg/controller/common/version/version.go

```diff
@@ -30,26 +32,27 @@ var (
 )
 
 // MinMaxVersion holds the minimum and maximum supported versions.
+// Could be replaced with semver.Range if we didn't have to support GlobalMinStackVersion.
```

Comment:
I think it would also no longer be possible to return detailed errors from `WithinRange` ?
Otherwise would it help to declare `GlobalMinStackVersion` as  type `semver.Range`:

```golang
	if ubiOnly {
		container.SetContainerSuffix("-ubi8")
		version.GlobalMinStackVersion = semver.MustParseRange(">7.10.0")
	}
```

And use the `AND` receiver ?

```golang
// CheckSupportedStackVersion checks that the given version is a valid Stack version supported by ECK.
func CheckSupportedStackVersion(proposed version.Version, supported semver.Range) field.ErrorList {
	if (version.GlobalMinStackVersion.AND(supported))(proposed) {
		return field.ErrorList{field.Invalid(field.NewPath("spec").Child("version"), proposed, fmt.Sprintf("Unsupported version: %s", proposed))}
	}
	return nil
}
```

Reply:
That is a clever idea. As you pointed out yourself, the downside is that the error message would be less detailed because `semver.Range` only gives an yes/no answer and no explanation. Therefore, I think I'd prefer to keep the existing implementation of `MinMaxVersion` for now because it is simple enough. If we need to support non-contiguous version ranges in the future, we can consider switching to `semver.Range`. WDYT?

Reply:
👍 , SGTM